ActionNetworkContactImporter.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+require_once("AbstractContactImporter.php");
+
+class ActionNetworkContactImporter extends AbstractContactImporter
```

Comment:
might name this OSDI vs Action Network? Since we're going for the more general case?

Reply:
I'm a little confused here -- AbstractContactImporter is in the context of OSDI, whereas ActionNetworkContactImporter  handles the specific ActionNetwork endpoint, right?

Reply:
Ah, got it -- I misread then. This maybe you want in the UI instead perhaps -- I imagine a user of the extension will get a settings page where they can enter their API key and paste in the OSDI endpoint? But that could be later work after this basic is done.

Reply:
That's reasonable. This class exclusively pulls contacts from the ActionNetwork API, so could we just assume that the endpoint is /people?  In any case, I agree we should throw the hardcoded endpoint ni a yaml file later to pull out.

Reply:
config file sounds like a good option, yeah!

Now, here's where HAL comes in. Really what you should need is simply the AEP's endpoint (https://actionnetwork.org/docs/v2/aep). Once you have that, you should be able to follow the link labeled `osdi:people` to find the correct person endpoint on that system.

So ideally, configuration is only "paste in the AEP URL" and you should be able to take it from there, in a generic way for all OSDI systems.

Reply:
sounds good ! What was that query language you mentioned earlier in the mattermost? I'm thinking i'd need it to insert a person by email, but I took a cursory look through the docs and wasn't able to find them.

Reply:
Query language is for GET, so you'll need it to, say, pull anyone new since the last sync: https://actionnetwork.org/docs/v2/#odata

For inserting a person, you want the person signup helper: https://actionnetwork.org/docs/v2/person_signup_helper

Most servers will either just use a matching strategy by default (like we do, we always match by email) or may allow some options there as defined by their docs. So you really don't need to insert a person by email, you just insert a person, and the server on the other end will handle matching.